### PR TITLE
Multiple documents update

### PR DESCRIPTION
**Pull request by pcmiel**

### CODE REVIEW

File: src/main/java/com/futureprocessing/documentjuggler/query/QueriedDocumentsImpl.java [hunk 1]
```diff
@@ -64,7 +64,7 @@ public MODEL first(String... fieldsToFetch) {
     @Override
     public UpdateResult update(UpdateConsumer<MODEL> consumer) {
         BasicDBObject document = updateProcessor.process(consumer);
-        WriteResult result = collection.update(query, document);
+        WriteResult result = collection.update(query, document, false, true);
```
Comment (pcmiel) [inline src/main/java/com/futureprocessing/documentjuggler/query/QueriedDocumentsImpl.java]: If no document updated, then result is equal to null.

File: src/test/java/com/futureprocessing/documentjuggler/UpdateTest.java [hunk 2]
```diff
@@ -38,7 +38,6 @@
     public void before() {
         given(db.getCollection(any())).willReturn(collection);
         given(collection.update(any(), any())).willReturn(writeResult);
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/UpdateTest.java]: This line requires changing. Problem is in wrong mocks.

File: src/test/java/com/futureprocessing/documentjuggler/UpdateTest.java [hunk 3]
```diff
@@ -56,18 +55,6 @@ public void shouldUpdateDocumentInCorrectCollection() {
 
 
     @Test
-    public void shouldReturnUpdateResult() {
```
Comment (szymonsobocik) [inline src/test/java/com/futureprocessing/documentjuggler/UpdateTest.java]: This test should not be deleted.

File: src/main/java/com/futureprocessing/documentjuggler/query/QueriedDocumentsImpl.java (hunk 1 above)
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/query/QueriedDocumentsImpl.java]: No, It always returns writeResult. Your problem lies in mocks.

File: src/main/java/com/futureprocessing/documentjuggler/update/OperationResult.java [hunk 4]
```diff
@@ -7,7 +7,7 @@
     private final int affectedCount;
 
     public OperationResult(WriteResult result) {
-        affectedCount = result.getN();
+        affectedCount = result != null ? result.getN() : 0;
```
Comment (szymonsobocik) [inline src/main/java/com/futureprocessing/documentjuggler/update/OperationResult.java]: This should not accept result as null.

**Comment by szymonsobocik** [comment]
+1

**Comment by exys666** [comment]
+1

**Comment by szymonsobocik** [comment]
Fixes #17

**Pull request merged**